src/library/Hooks/usePrices/index.tsx

```diff
@@ -52,20 +52,24 @@ export const usePrices = () => {
 
   // initial price subscribe
   useEffect(() => {
-    initiatePriceInterval();
-    return () => {
-      if (priceHandle !== null) {
-        clearInterval(priceHandle);
+    if (services.includes('binance_spot')) {
+      if (priceHandle === null) {
+        initiatePriceInterval();
       }
-    };
+    } else if (priceHandle !== null) {
+      clearInterval(priceHandle);
+    }
   }, []);
 
   // resubscribe on network toggle
   useEffect(() => {
-    if (priceHandle !== null) {
+    if (services.includes('binance_spot')) {
+      if (priceHandle === null) {
```

Comment:
Just to make sure. Should this check be here? I know the function inside is `initiatePriceInterval`, but it seems it was called always before

Reply:
Huh yeah, great spot. I don't remember why I left it like that, probably a copy-paste mistake. Btw I realized that those 2 `useEffect`s perform 2 separate jobs which, if merged into 1, a single `useEffect` is actually tailored for, so I merged those into one.